base/array.hpp

```diff
@@ -102,16 +130,33 @@ class BoundedArray final {
 using Bytes = Array<std::uint8_t>;
 using UniqueBytes = UniqueArray<std::uint8_t>;
 
+inline UniqueBytes SerializeAsBytes(
+    google::protobuf::MessageLite const& message);
```

Comment:
This doesn't seem to belong here.  Could we put it, e.g., in a `base/proto.hpp` or `base/serialization.hpp` header?